### PR TITLE
Consistent argument semantics in model diagram plot function calls

### LINKED ISSUE #1141: Consistent args for model diagram plot functions
## What and why
Currently one accepts entire args object and other one accepts a few specific arguments.

## Solution(s)
Both accept similar specific args rather than entire args object.

## Blocked by or pending

### PR DESCRIPTION
**Pull request by erikr**
## Issues closed
Closes #1141

## Changelog
- [x] Function which plots condensed architecture diagram is called with specific arguments instead of entire `args` object.
- [x] File names of architecture diagram images are `architecture` and `architecture-condensed` so they are sorted adjacent to each other in the `output_folder`.
- [x] File names of other plots use `-` instead of `_`.

## Checklist
- [x] The following are complete:
    - Added or revised tests if appropriate.
    - If new packages are installed, `requirements.txt` is updated.
    - If documentation should be updated, a new issue was made in `ml4c3-wiki`.
    - The PR name describes its content and does not contain "Ref #".
    - The PR does not add PHI.

### CODE REVIEW

**Comment by erikr** [comment]
@StevenSong when you have a chance, can you please review? Thanks!

File: ml4c3/arguments.py [hunk 1]
```diff
@@ -397,6 +397,7 @@ def parse_args() -> argparse.Namespace:
         nargs=4,
         default=["convolution", "normalization", "activation", "dropout"],
         choices=["convolution", "normalization", "activation", "dropout"],
+        type=list,
```
Comment (StevenSong) [inline ml4c3/arguments.py:400]: this is redundant with `nargs=4` akin to how we chose not to specify `type=bool` when `action="store_true/false"`

File: ml4c3/arguments.py [hunk 2]
```diff
@@ -425,12 +426,14 @@ def parse_args() -> argparse.Namespace:
     model_parser.add_argument(
         "--conv_padding",
         default="same",
+        type=str,
```
Comment (StevenSong) [inline ml4c3/arguments.py:429]: this is redundant as argument types are strings unless otherwise specified

File: ml4c3/arguments.py [hunk 3]
```diff
@@ -425,12 +426,14 @@ def parse_args() -> argparse.Namespace:
...
         help="Valid or same border padding on the convolutional layers.",
     )
     model_parser.add_argument(
         "--pool_type",
         nargs="?",
         choices=["max", "average"],
+        type=str,
```
Comment (StevenSong) [inline ml4c3/arguments.py:436]: redundant

File: ml4c3/models.py [hunk 4]
```diff
@@ -248,19 +247,38 @@ def sklearn_model_loss_from_dataset(
...
+                dense_layers=args.dense_layers,
+                image_ext=args.image_ext,
+                normalization_layer=args.normalization_layer,
+                output_folder=args.output_folder,
+                pool_type=args.pool_type,
+                residual_block_layer_order=args.residual_block_layer_order,
+                residual_blocks=args.residual_blocks,
+                spatial_dropout=args.spatial_dropout,
+                tensor_maps_in=args.tensor_maps_in,
+                tensor_maps_out=args.tensor_maps_out,
```
Comment (StevenSong) [inline ml4c3/models.py:276]: this is alphabetically sorted but not necessarily logically sorted,  e.g. I usually put the conv/dense/res block arguments together

style choice, fine either way

File: ml4c3/plots.py [hunk 5]
```diff
@@ -1222,14 +1224,14 @@ def _text_on_plot(axes, x, y, text, alpha=0.8, background="white"):
     t.set_bbox({"facecolor": background, "alpha": alpha, "edgecolor": background})
 
 
-def plot_architecture_diagram(dot: pydot.Dot, image_path: str):
```
Comment (StevenSong) [inline ml4c3/plots.py:1225]: pydot might be an unused import now, might be okay to remove

**Review by StevenSong — CHANGES_REQUESTED** [review]
great cleanup, minor requests

File: ml4c3/arguments.py (hunk 2 above)
Comment (erikr) [reply ml4c3/arguments.py:429]: Although it is redundant, I like it being explicit. I was adding type hints to the function and had to refer to `arguments.py` to figure out the types. But if args are default strings, then can remove.

File: ml4c3/models.py (hunk 4 above)
Comment (erikr) [reply ml4c3/models.py:276]: I used to logically sort but it became significant manual effort. `:sort` in vim does this instantly, and if we consistently sort in alphabetical order, it should be easier to find args.

File: ml4c3/plots.py (hunk 5 above)
Comment (erikr) [reply ml4c3/plots.py:1225]: We initialize a `pydot.Node` later on in the function so It think we still need to import it.

However I will only import what we use: `from pydot import Node`.

Comment (StevenSong) [reply ml4c3/plots.py:1225]: ah good point

**Review by StevenSong — APPROVED** [review]

**Pull request merged**